src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ExhaustivenessComputer.java

```diff
@@ -673,5 +782,432 @@ public String toString() {
+            }
+            Assert.check(index != (-1));
+
+            int indexFin = index;
+            Set<PatternDescription> filteredBasePatterns =
+                    basePatterns.stream()
+                                .filter(pd -> pd instanceof RecordPattern)
+                                .map(rp -> (RecordPattern) rp)
+                                .filter(rp -> types.isSameType(rp.recordType(), rootPatternRecord.recordType()))
+                                .map(rp -> rp.nested[indexFin])
+                                .collect(Collectors.toSet());
+
+            return basePatternsHaveRecordPatternOnThisSpot(filteredBasePatterns, rootPatternRecord.nested[index], added);
+        }
+
+        private boolean isUnderRoot(PatternDescription root, PatternDescription searchFor) {
+            if (root == searchFor) {
+                return true;
+            } else if (root instanceof RecordPattern rp) {
+                for (int c = 0; c < rp.nested.length; c++) {
+                    if (isUnderRoot(rp.nested[c], searchFor)) {
+                        return true;
+                    }
+                }
+            }
+            return false;
+        }
+
+    private Set<PatternDescription> joinSets(Collection<? extends PatternDescription> s1,
+                                             Collection<? extends PatternDescription> s2) {
+        Set<PatternDescription> result = new HashSet<>();
+
+        result.addAll(s1);
+        result.addAll(s2);
+
+        return result;
+    }
+
+    /*
+     * Based on {@code basePattern} generate new {@code RecordPattern}s such that all
+     * components instead of {@code replaceComponent}th component, which is replaced
+     * with values from {@code updatedNestedPatterns}. Resulting {@code RecordPatterns}s
+     * are sent to {@code target}.
+     */
+    private void generatePatternsWithReplacedNestedPattern(RecordPattern basePattern,
+                                                           int replaceComponent,
+                                                           Iterable<? extends PatternDescription> updatedNestedPatterns,
+                                                           Set<PatternDescription> sourcePatterns,
+                                                           Consumer<RecordPattern> target) {
+        for (PatternDescription nested : updatedNestedPatterns) {
+            PatternDescription[] newNested =
+                    Arrays.copyOf(basePattern.nested, basePattern.nested.length);
+            newNested[replaceComponent] = nested;
+            target.accept(new RecordPattern(basePattern.recordType(),
+                                            basePattern.fullComponentTypes(),
+                                            newNested,
+                                            sourcePatterns));
+        }
+    }
+
+    /* The stricness of determining the equivalent of patterns, used in
+     * nestedComponentsEquivalent.
```

Comment:
Also in `computeCoverage`. Can you provide an example/sentence of two strictly equivalent patterns and two loosely equivalent?